### PR TITLE
Fix error with Yaml library while using ML-PACE

### PR DESCRIPTION
**Pull request by srmnitc**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by srmnitc** [comment]
@conda-forge-admin, please rerender

**Comment by srmnitc** [comment]
> > * [ ]   Bumped the build number (if the version is unchanged)
> 
> Please bump the version number

Will do, thanks for the pointer. There seems to be an error:

```
File "/home/menon/miniconda3/envs/myenv/lib/python3.9/ctypes/__init__.py", line 374, in __init__
    self._handle = _dlopen(self._name, mode)
OSError: libyaml-cpp-pace.so.0.6: cannot open shared object file: No such file or directory
```
I am trying to fix this, in addition add tests for the library compilation. Feel free to add commits if you see anything strange

**Comment by srmnitc** [comment]
@jan-janssen The OSX builds seem to be failing, do you have an idea what could be the issue?

**Comment by jan-janssen** [comment]
```
OSError: dlopen(/Users/runner/miniforge3/conda-bld/lammps_1653857816595/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.7/site-packages/lammps/liblammps.dylib, 10): Library not loaded: @rpath/libyaml-cpp-pace.0.6.dylib
  Referenced from: /Users/runner/miniforge3/conda-bld/lammps_1653857816595/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeho/lib/python3.7/site-packages/lammps/liblammps.dylib
```
It seems like the link to the pace library is incorrect. Are you sure it is compiled as `*.dylib` vs. `*.a` ?

File: recipe/build.sh [hunk 1]
```diff
@@ -42,9 +42,13 @@ cd build_lib
 cmake -D BUILD_LIB=ON -D BUILD_SHARED_LIBS=ON -D BUILD_MPI=ON -D PKG_MPIIO=ON -D LAMMPS_EXCEPTIONS=yes $args ../cmake
 make # -j${NUM_CPUS}
 cp liblammps${SHLIB_EXT}* ../src  # For compatibility with the original make system.
+cp liblammps_pace.* ../src
+cp build-yaml-cpp/libyaml-cpp-pace${SHLIB_EXT}* ../src
```
Comment (jan-janssen) [inline recipe/build.sh:46]: This part should be done automatically in the build script - otherwise the pace library interface is most likely broken upstream.

**Comment by jan-janssen** [comment]
I marked the old packages as broken https://github.com/conda-forge/admin-requests/pull/443

**Comment by jan-janssen** [comment]
@srmnitc Is the `libyamp-cpp-pace` different from https://anaconda.org/conda-forge/yaml/ ?

**Comment by srmnitc** [comment]
> @srmnitc Is the `libyamp-cpp-pace` different from https://anaconda.org/conda-forge/yaml/ ?

Yes, I think so. But the yaml library is compiled into a different namespace in the ACE code; based on some discussions. So I do not know if installing from conda-forge will solve the issue. See [here](https://github.com/lammps/lammps/pull/2968).

**Comment by akohlmey** [comment]
This was fixed in LAMMPS upstream commit https://github.com/lammps/lammps/commit/056fd8be8f34b2c9f141089ddaca9de29c046072

**Comment by srmnitc** [comment]
@jan-janssen The issue was fixed in a new LAMMPS release. I am trying to update the recipe. Could you please take a look and see what could be failing?

**Comment by jan-janssen** [comment]
@srmnitc I updated lammps to the latest version, hopefully this also resolves the ML-PACE issue.

**Comment by srmnitc** [comment]
> @srmnitc I updated lammps to the latest version, hopefully this also resolves the ML-PACE issue.

Thanks! The tests are running now, so we should be good to go.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -67,6 +67,7 @@ requirements:
     - python
     - setuptools
     - {{ mpi }}
+    - openssh
```
Comment (jan-janssen) [inline recipe/meta.yaml:70]: I do not get it, the only thing that changed is adding `openssh` ? How is this related to `ML-PACE` ?

Comment (srmnitc) [reply recipe/meta.yaml:70]: I added openssh to fix some other error, which I cant find now since the logs expired. I will remove it and see if there are any issues.

Comment (jan-janssen) [reply recipe/meta.yaml]: These changes are already included in the current `main` branch https://github.com/conda-forge/lammps-feedstock/blob/main/recipe/build.sh

Comment (srmnitc) [reply recipe/meta.yaml]: I am sure I follow, it seems that ML-PACE is turned off currently in the main branch. Or am I missing something?

Comment (jan-janssen) [reply recipe/meta.yaml]: Looking at the files in this pull request, you do not change the `build.sh` script just the `run.py` and the `meta.yaml` files are changed. So it seems this pull request does not enable the `ML-PACE` package either.

Comment (srmnitc) [reply recipe/meta.yaml]: Ok, thanks, I see the point now. Shall I simply close this and open a new pull request with ML-PACE on and we see if everything works?
